### PR TITLE
Replace blockchain db lock to tokio::sync:rwlock

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description

Replaces the std::sync:rwlock of the block chain db with a tokio::sync:rwlock

## Motivation and Context

We have encountered some problems with std::sync::rwlock with async code. I have replaced it with the tokio::sync::rwlock. It should provide more stability. 

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by neonknight64 — DISMISSED** [review]
LGTM

File: base_layer/core/src/chain_storage/blockchain_database.rs [hunk 1]
```diff
@@ -281,24 +280,12 @@ where T: BlockchainBackend
...
-    pub fn db_read_access(&self) -> Result<RwLockReadGuard<T>, ChainStorageError> {
-        self.db.read().map_err(|e| {
-            error!(
-                target: LOG_TARGET,
-                "An attempt to get a read lock on the blockchain backend failed. {:?}", e
-            );
-            ChainStorageError::AccessError("Read lock on blockchain backend failed".into())
-        })
+    pub fn db_read_access(&self) -> Result<tokio::sync::RwLockReadGuard<T>, ChainStorageError> {
+        Ok(block_on(async { self.db.read().await }))
```
Comment (sdbondi) [inline base_layer/core/src/chain_storage/blockchain_database.rs:284]: This is being spawned on a future-rs runtime.

**Review by sdbondi — COMMENTED** [review]
Using an async lock really only benefits if you are in an async context. `block_on` from futures-rs will initialise and use `LocalPool` which uses the current thread to do it's work, that's fine but it carries some overhead.  If we needed to go the async lock route for a specific reason, we should change the code to be async and run on the tokio runtime.

**Comment by SWvheerden** [comment]
I am closing this, because after further investigation, this is not the issue, and I doubt this will fix it.

**Pull request closed without merging**